x-pack/plugins/triggers_actions_ui/public/application/hooks/use_load_rule_alerts_aggregations.ts

```diff
@@ -146,51 +142,109 @@ export async function fetchRuleAlertsAggByTimeRange({
+                totalActiveAlerts: {
+                  term: {
+                    'kibana.alert.status': 'active',
+                  },
+                },
+                totalRecoveredAlerts: {
+                  term: {
+                    'kibana.alert.status': 'recovered',
+                  },
+                },
+              },
+            },
+          },
+          statusPerDay: {
+            date_histogram: {
+              field: '@timestamp',
+              fixed_interval: '1d',
+              extended_bounds: {
+                min: 'now-30d',
+                max: 'now',
               },
             },
             aggs: {
-              status_per_day: {
-                date_histogram: {
-                  field: '@timestamp',
-                  fixed_interval: '1d',
+              alertStatus: {
+                terms: {
+                  field: 'kibana.alert.status',
                 },
               },
             },
           },
         },
       }),
     });
-    const active = res?.aggregations?.filterAggs.buckets.alert_active?.doc_count ?? 0;
-    const recovered = res?.aggregations?.filterAggs.buckets.alert_recovered?.doc_count ?? 0;
+
+    const active = res?.aggregations?.total.buckets.totalActiveAlerts?.doc_count ?? 0;
+    const recovered = res?.aggregations?.total.buckets.totalRecoveredAlerts?.doc_count ?? 0;
+    // Total alerts count in the last 30 days per status
+    const totalAlerts = active + recovered;
+
     const alertsChartData = [
-      ...res?.aggregations?.filterAggs.buckets.alert_active.status_per_day.buckets.map(
-        (bucket: BucketAggsPerDay) => ({
-          date: bucket.key_as_string,
-          status: 'active',
-          count: bucket.doc_count,
-        })
-      ),
-      ...res?.aggregations?.filterAggs.buckets.alert_recovered.status_per_day.buckets.map(
-        (bucket: BucketAggsPerDay) => ({
-          date: bucket.key_as_string,
-          status: 'recovered',
-          count: bucket.doc_count,
-        })
+      ...res?.aggregations?.statusPerDay.buckets.reduce(
+        (
+          acc: AlertChartData[],
+          dayAlerts: {
+            key_as_string: string;
+            doc_count: number;
+            alertStatus: {
+              buckets: Array<{
+                key: string;
+                doc_count: number;
+              }>;
+            };
+          }
+        ) => {
+          // We are adding this to each day to construct the 30 days bars (background bar) when there is no data for a given day or to show the delta today alerts/total alerts.
+          const totalDayAlerts = {
+            date: dayAlerts.key_as_string,
+            count: totalAlerts,
+            status: 'total',
+          };
+
+          if (dayAlerts.doc_count > 0) {
+            // If there are alerts in this day, we construct the chart data.
+            return [
+              ...acc,
+              ...dayAlerts.alertStatus.buckets.map((alert) => ({
```

Comment:
Tell me if I am wrong here, but I feel that you should also add the totalDayAlert too so you can avoid the white background when you have data.

<img width="29" alt="image" src="https://user-images.githubusercontent.com/189600/181758003-93bf8013-ddae-4983-b704-10b7878150d0.png">

Reply:
@XavierM, excellent point. Actually, `totalDayAlert` was added to each day and calculated based on the number of alerts for that day. 
But then once the 2nd of alerts kicked in. The chart flipped upside down means the `totalDayAlert` (grey color) becomes at the bottom and the recovered and active at the top. I tried to fix that in many ways this morning but with no success and I don't want to spend more time on it as we all agreed. 

BTW, when we hover over the bar the gray bar appears again. 

I think this limitation is ok for this version, what do you think @maciejforcone?

Reply:
I think we can do it just like that -> https://github.com/fkanout/kibana/pull/3/files

Reply:
Thanks, @XavierM. But it has exactly the same issue I mentioned -the chart is _flipped_. You need at least two days of data to see the issue.

<img width="602" alt="Screenshot 2022-07-29 at 17 11 28" src="https://user-images.githubusercontent.com/6838659/181791150-512f6369-a76c-4ccc-a7ec-efba32676f90.png">

Here is [my commit](https://github.com/elastic/kibana/pull/137476/commits/2720e2ebce137521f3a5d2030720f4a738623888#diff-74f788624580e7eeb63a80c46b40834c7448c638c12c4728907fc74e70cec979L244) where I deleted my solution after I saw the issue

Reply:
I failed.... 😞 but I was also looking everywhere in kibana, and it looks like that nobody is doing the fake background like we do, maybe for good reason. I am wondering if we should remove it because if we are having alerts/data, we most likely not going to see this grey background anyway. It is better to be simplistic sometime.

Reply:
@XavierM No worries! BTW, the only way I found to make it works is by making chart reading starting from the left, which means the today bar will be at the very left bar. 

To answer your question. I don't mind removing the background/reference bars.

Reply:
Alright, after talking to @angorayc, we found a cool solution about it and now she might have to fix some other graph in security solution LOL.

The [PR](https://github.com/fkanout/kibana/pull/3/files) is re-open @fkanout